integration_tests/src/main/python/iceberg_test.py

```diff
@@ -95,6 +97,8 @@ def setup_iceberg_table(spark):
 @ignore_order(local=True) # Iceberg plans with a thread pool and is not deterministic in file ordering
 @pytest.mark.parametrize("data_gens", iceberg_gens_list, ids=idfn)
 @pytest.mark.parametrize('reader_type', rapids_reader_types)
+# TODO: Add support for nested data type: https://github.com/NVIDIA/spark-rapids/issues/12298
```

Comment:
Please split this case for non-nested data_gens without `allow_non_gpu` to make sure the non-nested types works.